### PR TITLE
Added new sample for storage to upload chunked stream

### PR DESCRIPTION
**Pull request by vishwarajanand**
Per https://github.com/googleapis/google-cloud-php/issues/4155, our users are requesting a sample / documentation for uploading objects of a fixed chunk sizes to Storage. Existing samples cover file/memory uploads in entirety but not with controlled chunk sizes.

**NOTE:** This change will replace the samples already published [on this devsite link](https://cloud.google.com/storage/docs/samples/storage-stream-file-upload#storage_stream_file_upload-php).

## Changes
1. Added a new sample: `storage_stream_file_upload`
2. Added test for `testUploadAndDownloadObjectStream`
3. Removed sample tag from older sample to upload files: `upload_object.php`

## Tests

1. Uploaded a local file by executing sample:

```
➜  storage git:(upload_chunked_stream) ✗ php src/upload_object_stream.php $GOOGLE_STORAGE_BUCKET "stream-$(date +%F-%Ih-%Mm-%Ss)" "abcdedghijklmnopqrstuvwxyz"                                                                
Uploaded abcdedghijklmnopqrstuvwxyz to gs://thebookclub/stream-2022-06-17-10h-14m-54s
➜  storage git:(upload_chunked_stream)
```

2. Test run:

```
➜  storage git:(upload_chunked_stream) ✗ XDEBUG_MODE=coverage ../testing/vendor/bin/phpunit --verbose -c phpunit.xml.dist test/ObjectsTest.php                                             
PHPUnit 8.5.23 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.1.2 with Xdebug 3.1.2
Configuration: /usr/local/google/home/vishwarajanand/github/php-docs-samples/storage/phpunit.xml.dist
Error:         This version of PHPUnit does not support code coverage on PHP 8

..........                                                        10 / 10 (100%)

Time: 3.42 seconds, Memory: 16.00 MB

OK (10 tests, 44 assertions)
➜  storage git:(upload_chunked_stream)
```

### CODE REVIEW

File: storage/src/upload_object_stream.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+use Google\Cloud\Storage\WriteStream;
+
+/**
+ * Upload a chunked file stream.
+ *
+ * @param string $bucketName The name of your Cloud Storage bucket.
+ * @param string $objectName The name of your Cloud Storage object.
+ * @param string $source The path to the file to upload.
+ */
+function upload_object_stream($bucketName, $objectName, $source)
```
Comment (frankyn) [inline storage/src/upload_object_stream.php:37]: Is there an equivalent ReadStream API for PHP?

File: storage/src/upload_object_stream.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    $storage = new StorageClient();
+    $bucket = $storage->bucket($bucketName);
+    $writeStream = new WriteStream(null, [
+        'chunkSize' => 1024 * 256, // 256KB
+    ]);
+    $uploader = $bucket->getStreamableUploader($writeStream, [
+        'name' => $objectName
+    ]);
+    $writeStream->setUploader($uploader);
+    $file = fopen($source, 'r');
```
Comment (frankyn) [inline storage/src/upload_object_stream.php:52]: You don't necessarily need to show opening a file to read from, you could create random data in a loop to demonstrate streaming writes which can be easily adapted to any use-case.

Comment (vishwarajanand) [reply storage/src/upload_object_stream.php]: I have changed the code to receive a string to be a little more generic, using streams directly as params will make it very complicated as I will have to make the stream pass by reference and will lose the ability to run the sample as `Test#1` in description of this PR.

**Comment by vishwarajanand** [comment]
Seems like unrelated `HMACTest.php` are failing, will check them later.

```
1) Google\Cloud\Samples\Storage\Tests\HmacTest::testHmacKeyList
Google\Cloud\Core\Exception\FailedPreconditionException: {"error":{"code":412,"message":"Request violates constraint 'constraints/iam.disableServiceAccountKeyCreation'","errors":[{"message":"Request violates constraint 'constraints/iam.disableServiceAccountKeyCreation'","domain":"global","reason":"conditionNotMet","locationType":"header","location":"If-Match"}]}}

/tmpfs/src/github/php-docs-samples/storage/vendor/google/cloud-core/src/RequestWrapper.php:368
/tmpfs/src/github/php-docs-samples/storage/vendor/google/cloud-core/src/RequestWrapper.php:207
/tmpfs/src/github/php-docs-samples/storage/vendor/google/cloud-core/src/RestTrait.php:97
/tmpfs/src/github/php-docs-samples/storage/vendor/google/cloud-storage/src/Connection/Rest.php:442
/tmpfs/src/github/php-docs-samples/storage/vendor/google/cloud-storage/src/StorageClient.php:556
/tmpfs/src/github/php-docs-samples/storage/test/HmacTest.php:42

```

File: storage/src/upload_object_stream.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+use Google\Cloud\Storage\WriteStream;
+
+/**
+ * Upload a chunked file stream.
+ *
+ * @param string $bucketName The name of your Cloud Storage bucket.
+ * @param string $objectName The name of your Cloud Storage object.
+ * @param string $contents The contents to upload via stream chunks.
+ */
+function upload_object_stream($bucketName, $objectName, $contents)
```
Comment (bshaffer) [inline storage/src/upload_object_stream.php:37]: nit: add typehints!
```suggestion
function upload_object_stream(string $bucketName, string $objectName, string $contents): void
```

File: storage/src/upload_object_stream.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    while (($line = stream_get_line($stream, 1024 * 256)) !== false) {
+        $writeStream->write($line);
+    }
+    $writeStream->close();
+
+    printf('Uploaded %s to gs://%s/%s' . PHP_EOL, $contents, $bucketName, $objectName);
+}
+# [END storage_stream_file_upload]
+
+// The following 2 lines are only needed to run the samples
```
Comment (bshaffer) [inline storage/src/upload_object_stream.php:62]: super-nit - let's make this more descriptive
```suggestion
// The following 2 lines are only needed to run the samples from the CLI
```

**Review by bshaffer — COMMENTED** [review]
This looks great! Just two nits!

**Review by frankyn — APPROVED** [review]

File: storage/src/upload_object_stream.php (hunk 3 above)
Comment (vishwarajanand) [reply storage/src/upload_object_stream.php]: God bless your 👀

**Comment by vishwarajanand** [comment]
Reported an issue for Snippet Bot for not checking already added branch(commit SHA) when removing region tags: https://github.com/googleapis/repo-automation-bots/issues/4025

**Review by bshaffer — APPROVED** [review]

**Pull request merged**